test/ooni/bouncer.cpp

```diff
@@ -0,0 +1,85 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#define CATCH_CONFIG_MAIN
+
+#include "../src/libmeasurement_kit/ext/catch.hpp"
+
+#include "../src/libmeasurement_kit/ooni/bouncer_impl.hpp"
+
+using namespace mk;
+using namespace mk::ooni;
+using namespace mk::ooni::bouncer;
+
+void request_invalid(Settings, http::Headers, std::string,
+                     Callback<Error, Var<http::Response>> cb,
+                     Var<Reactor> = Reactor::global(),
+                     Var<Logger> = Logger::global(),
+                     Var<http::Response> previous = nullptr,
+                     int nredirects = 0) {
+    Var<http::Response> response(new http::Response());
+    response->body = "{\"error\": \"invalid-request\"}";
+    cb(NoError(), response);
+}
+
+TEST_CASE("The bouncer can handle an invalid request") {
+    loop_with_initial_event([]() {
+        post_net_tests_impl<request_invalid>(
+            "https://a.collector.ooni.io/bouncer/net-tests", "web-connectivity",
```

Comment:
I was initially confused by this unittest because I didn't notice the mocked out `request_invalid` and the request actually looked valid to me.

Maybe what you could be done is mark the request in some more obvious way (note the backend will reply with request invalid when it's not JSON).